src/main/java/darkpool/util/Storage.java

```diff
@@ -0,0 +1,127 @@
+package darkpool.util;
+
+import darkpool.task.Deadline;
+import darkpool.task.Event;
+import darkpool.task.Task;
+import darkpool.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Handles the loading and saving of tasks to and from a file.
+ */
+public class Storage {
+
+    private final String filePath;
+
+    /**
+     * Constructs a Storage object with the specified file path.
+     *
+     * @param filePath The path to the file where tasks are stored.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Checks if the file and its directory exist, and creates them if they do not.
+     *
+     * @return The file object representing the data file.
+     * @throws DarkpoolException If an I/O error occurs.
+     */
+    private File fileAndDirCheck() throws DarkpoolException {
```

Comment:
I think the name of this method can be changed to fit what W4.6b recommends.
![image](https://github.com/user-attachments/assets/fb508669-48c7-431e-89d1-aabbc79a8caf)

no hard feelings, just tryna get those participation marks yk

Reply:
Thanks for pointing the mistake out! I have fixed it.

Reply:
I have updated it to validateData. Thanks for pointing it out.